### PR TITLE
#23: changed values for binary parameters (boolean instead of integers)

### PR DESCRIPTION
**Pull request by slvnwhrl**

### CODE REVIEW

File: api/code/flaskr/model.py [hunk 1]
```diff
@@ -64,9 +64,14 @@ def validate_prediction_request(request):
             features.append(float(el))
 
         elif t == 'binary':
-            if not el.isdigit() or int(el) not in vals:
+            try:
+                el_parsed = json.loads(el)
+            except json.decoder.JSONDecodeError:
```
Comment (mialbu) [inline api/code/flaskr/model.py:69]: Please add tests for those new lines.

File: api/code/flaskr/model.py [hunk 2]
```diff
@@ -64,9 +64,14 @@ def validate_prediction_request(request):
...
                 return {'error': {'code': 400, 'msg': f'unallowed value for {field}'}}
-            features.append(int(el))
+                
+            if not isinstance(el_parsed, bool):
+                return {'error': {'code': 400, 'msg': f'unallowed value for {field}'}}
```
Comment (mialbu) [inline api/code/flaskr/model.py:73]: Please add tests for those lines as well.

File: api/tests/test_model.py [hunk 3]
```diff
@@ -86,17 +87,16 @@ def valid_model_input():
...
         validated = model.validate_prediction_request(request)
-        print(validated)
         assert ('error' in validated and validated['error'][
             'msg'] == 'Make sure all required fields are submitted') is not True
 
 
 def test_model_input_validation_valid_response(flask_app, valid_model_input):
     with flask_app.test_request_context(data=valid_model_input):
         validated = model.validate_prediction_request(request)
+        print(validated)
```
Comment (mialbu) [inline api/tests/test_model.py:97]: Remove this print.

**Review by mialbu — CHANGES_REQUESTED** [review]
Good stuff. Only minor requested changes.

File: api/tests/test_model.py (hunk 3 above)
Comment (slvnwhrl) [reply api/tests/test_model.py]: Removed it

File: api/code/flaskr/model.py (hunk 2 above)
Comment (slvnwhrl) [reply api/code/flaskr/model.py:73]: Done!

File: api/code/flaskr/model.py (hunk 1 above)
Comment (slvnwhrl) [reply api/code/flaskr/model.py:69]: Done!

**Pull request merged**